modules/rss.py

```diff
@@ -37,14 +31,22 @@ def _format_entry(self, server, channel, feed_title, entry, shorten):
                 pass
         link = "%s" % link if link else ""
 
-        feed_title_str = "%s" % feed_title if feed_title else ""
+        variables = dict(
+            longtitle=feed_title or "",
+            title=utils.parse.line_normalise(utils.http.strip_html(
+                entry["title"])),
+            link=link or "",
+            author=entry.get("author", "unknown author") or "",
+        )
+
         # just in case the format starts keyerroring and you're not sure why
         self.log.trace("RSS Entry: " + str(entry))
         try:
-            format = channel.get_setting("rss-format", "$longtitle: $title by $author - $link").replace("$longtitle", feed_title_str).replace("$title", title).replace("$link", link).replace("$author", author).format(**entry)
+            template = channel.get_setting("rss-format", "$longtitle: $title by $author - $link")
+            format = format_token_replace(template, variables).format(**entry)
         except KeyError:
             self.log.warn(f"Failed to format RSS entry for {channel}. Falling back to default format.")
-            format = f"{feed_title_str}: {title} by {author} - {link}"
+            format = "{longtitle}: {title} by {author} - {link}".format(**variables)
```

Comment:
This entire check is now unnecessary since `format_token_replace()` will silently fail to convert any unknown placeholders.